File: australia_map.py
states = ['WA','NT','SA','QLD','NSW','VIC','T']

neighbors = {
    'WA': ['NT','SA'],
    'NT': ['WA','SA','QLD'],
    'SA': ['WA','NT','QLD','NSW','VIC'],
    'QLD': ['NT','SA','NSW'],
    'NSW': ['QLD','SA','VIC'],
    'VIC': ['SA','NSW'],
    'T': []
}

colors = ['Red','Green','Blue']

def is_valid(state, color, assignment):
    for neighbor in neighbors[state]:
        if neighbor in assignment and assignment[neighbor] == color:
            return False
    return True

def backtrack(assignment):
    if len(assignment) == len(states):
        return assignment

    for state in states:
        if state not in assignment:
            for color in colors:
                if is_valid(state, color, assignment):
                    assignment[state] = color
                    result = backtrack(assignment)
                    if result:
                        return result
                    del assignment[state]
            return None

solution = backtrack({})
print("Australia Map Coloring:", solution)

File: australia_map.c
#include <stdio.h>
#include <stdbool.h>

char *states[] = {"WA","NT","SA","QLD","NSW","VIC","T"};
char *colors[] = {"Red","Green","Blue"};

int graph[7][7] = {
    {0,1,1,0,0,0,0},
    {1,0,1,1,0,0,0},
    {1,1,0,1,1,1,0},
    {0,1,1,0,1,0,0},
    {0,0,1,1,0,1,0},
    {0,0,1,0,1,0,0},
    {0,0,0,0,0,0,0}
};

int solution[7];

bool isSafe(int node, int color) {
    for(int i=0;i<7;i++) {
        if(graph[node][i] && solution[i]==color)
            return false;
    }
    return true;
}

bool solve(int node) {
    if(node==7) return true;

    for(int c=1;c<=3;c++) {
        if(isSafe(node,c)) {
            solution[node]=c;
            if(solve(node+1)) return true;
            solution[node]=0;
        }
    }
    return false;
}

int main() {
    solve(0);
    for(int i=0;i<7;i++) {
        printf("%s -> %s\n", states[i], colors[solution[i]-1]);
    }
    return 0;
}
